crates/router/src/core/webhooks.rs

```diff
@@ -457,24 +458,22 @@ async fn bank_transfer_webhook_flow<W: types::OutgoingWebhookType>(
                 .change_context(errors::ApiErrorResponse::WebhookProcessingFailure)
                 .attach_printable("did not receive payment id from payments core response")?;
 
-            let event_type: enums::EventType = payments_response
-                .status
-                .foreign_try_into()
-                .into_report()
-                .change_context(errors::ApiErrorResponse::WebhookProcessingFailure)
-                .attach_printable("error mapping payments response status to event type")?;
-
-            create_event_and_trigger_outgoing_webhook::<W>(
-                state,
-                merchant_account,
-                event_type,
-                enums::EventClass::Payments,
-                None,
-                payment_id,
-                enums::EventObjectType::PaymentDetails,
-                api::OutgoingWebhookContent::PaymentDetails(payments_response),
-            )
-            .await?;
+            let event_type: enums::EventType = payments_response.status.foreign_into();
+
+            // If event is NOT an UnsupportedEvent, trigger Outgoing Webhook
+            if !matches!(event_type, enums::EventType::UnsupportedEvent) {
+                create_event_and_trigger_outgoing_webhook::<W>(
```

Comment:
also, can't we move this logic inside the function
`create_event_and_trigger_outgoing_webhook`.

That would reduce replication and be a little fault tolerant.
what do you think?

Reply:
For all the Webhooks flows this common function is being called, it might affect the flexibility of the function, Currently not updating it, if required can be taken up in refactoring